### PR TITLE
refactor: clean up [Dune_rpc_server] api

### PR DESCRIPTION
**Pull request by rgrinberg**
* Add missing documentation
* Remove useless functions from the public API
* Remove pointless one liner helpers. No idea why I added those in the first place, but they're just making the code hard to follow.

### CODE REVIEW

File: src/dune_rpc_server/dune_rpc_server.mli [hunk 1]
```diff
@@ -127,11 +115,15 @@ module Handler : sig
...
+
+    type poller
+
     val implement_poll :
          's t
       -> 'a Procedures.Poll.t
-      -> on_poll:('s Session.t -> Poller.t -> 'a option Fiber.t)
-      -> on_cancel:('s Session.t -> Poller.t -> unit Fiber.t)
+      -> on_poll:('s Session.t -> poller -> 'a option Fiber.t)
+      -> on_cancel:('s Session.t -> poller -> unit Fiber.t)
```
Comment (snowleopard) [inline src/dune_rpc_server/dune_rpc_server.mli:126]: This looks odd. What can you do with the `poller` if it's abstract?

Comment (rgrinberg) [reply src/dune_rpc_server/dune_rpc_server.mli:126]: Nothing. This function is only used for tests and the tests don't care about the poller argument. I could specialize this function for the test and omit this argument.

File: src/dune_rpc_server/dune_rpc_server.mli [hunk 2]
```diff
@@ -127,11 +115,15 @@ module Handler : sig
     -> unit
 
   module Private : sig
+    (** Raw long polling machinery only good for tests *)
```
[the remark this answers is not in the document]
Comment (rgrinberg) [reply src/dune_rpc_server/dune_rpc_server.mli:116]: done.

File: src/dune_rpc_server/dune_rpc_server.mli (hunk 1 above)
Comment (snowleopard) [reply src/dune_rpc_server/dune_rpc_server.mli:124]: Yeah, I think we should drop this argument then, it's pretty confusing.

Comment (rgrinberg) [reply src/dune_rpc_server/dune_rpc_server.mli]: Done.

**Review by snowleopard — APPROVED** [review]
Thanks!
